src/helpers/mod.rs

```diff
@@ -1,11 +1,58 @@
 pub mod error;
+pub mod mesh;
+#[cfg(test)]
+pub mod mock;
 pub mod models;
-pub mod ring;
 
 /// Represents a unique identity of each helper running MPC computation.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[derive(Copy, Clone, Debug, PartialEq, Hash, Eq)]
 pub enum Identity {
     H1,
     H2,
     H3,
 }
+
+#[derive(Debug, Copy, Clone)]
+pub enum Direction {
+    Left,
+    Right,
+}
+
+impl Identity {
+    #[must_use]
+    pub fn all_variants() -> &'static [Identity; 3] {
```

Comment:
what's the difference between this and something like

```rust
const all_variants: &'static [Identity; 3] = &[Identity::H1, Identity::H2, Identity::H3];
```

Reply:
oh you know i had no idea godbolt supported rust. cool!

Reply:
you know what, there actually is a difference. there isn't a way to reference `const all_variants` without a reference to a particular instance, i.e. `self.all_variants`. while the `fn all_variants()` can be referenced via `Identity::all_variants()`

Reply:
hmm, does the following cover this use case? 

```rust
impl Identity {
    pub const ALL_VARIANTS: &'static [Identity; 3] = &[Identity::H1, Identity::H2, Identity::H3];
   
    pub fn l(&self) -> usize {
        Self::ALL_VARIANTS.len()
    }
}

fn l() -> usize {
    Identity::ALL_VARIANTS.len()
}


```

Reply:
oh, yeah. so it does work. my IDE was telling me that was invalid, but i didn't actually test it on compile. thanks CLion